web/files/referencedby.go

```diff
@@ -10,6 +11,53 @@ import (
 	"github.com/labstack/echo/v4"
 )
 
+// ContentType is the official mime-type for JSON-API
+const ContentType = "application/vnd.api+json"
+
+// UpdatedReferences is a container for the updated list of referenced_by object
+// for an io.cozy.files document.
+type UpdatedReferences struct {
+	DocRev       string                 `json:"rev"`
+	ReferencedBy []couchdb.DocReference `json:"referencedBy"`
+}
+
+func NewUpdatedReferences(rev string, refs []couchdb.DocReference) UpdatedReferences {
+	u := UpdatedReferences{DocRev: rev}
+	u.ReferencedBy = make([]couchdb.DocReference, len(refs))
+	copy(u.ReferencedBy, refs)
+
+	return u
+}
+
+// MarshalUpdatedReferences serializes an UpdatedReferences to JSON.
+// It returns a json.RawMessage that can be returned in an HTTP response.
+func MarshalUpdatedReferences(r UpdatedReferences) (json.RawMessage, error) {
+	b, err := json.Marshal(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// WriteData can be called to write an answer with a JSON document
+// containing a single UpdatedReferences object as data into an io.Writer.
+func WriteData(c echo.Context, statusCode int, r UpdatedReferences) error {
```

Comment:
`WriteData` is a bit too generic for naming a function inside the `files` package for work only for references. Maybe, we can call it `WriteReferences`.